packages/block-library/src/template-part/edit/placeholder/index.js

```diff
@@ -61,8 +74,15 @@ export default function TemplatePartPlaceholder( {
 		<>
 			{ step === PLACEHOLDER_STEPS.initial && (
 				<Placeholder
-					icon={ blockDefault }
-					label={ __( 'Template Part' ) }
+					icon={
+						areasByName[ area ]?.icon ||
+						areasByName.uncategorized.icon
+					}
+					label={
+						areasByName[ area ]?.label
+							? areasByName[ area ].label
+							: __( 'Template Part' )
+					}
```

Comment:
How about doing this logic in the select hook as well?
Thinking something like this:

```js
import { find } from 'lodash';

// ...

const { areaIcon, areaLabel } = useSelect(
	( select ) => {
		const definedAreas = select( editorStore ).__experimentalGetDefaultTemplatePartAreas();

		const selectedArea = find( definedAreas, { area } );
		const defaultArea = find( definedAreas, { area: 'uncategorized' } );

		return {
			areaIcon: selectedArea?.icon || defaultArea?.icon,
			areaLabel: selectedArea?.label || __( 'Template Part' ),
		};
	},
	[ area ]
);
```

Reply:
Great idea!